### ISSUE TITLE
Error extension name with --hpy-abi=universal

### ISSUE THREAD

**Issue by paugier** [hpyproject/hpy#103]
I try to build piconumpy with the flag `--hpy-abi=universal`. I use `python setup.py --hpy-abi=universal build_ext -if` and I get the following error:

```
running build_ext
building 'piconumpy._piconumpy_cpython_capi' extension
C compiler: gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC

creating build
creating build/temp.linux-x86_64-3.8
creating build/temp.linux-x86_64-3.8/piconumpy
compile options: '-I/home/pierre/.pyenv/versions/3.8.2/include/python3.8 -c'
extra options: '-Wfatal-errors -Werror'
gcc: piconumpy/_piconumpy_cpython_capi.c
creating build/lib.linux-x86_64-3.8
creating build/lib.linux-x86_64-3.8/piconumpy
gcc -pthread -shared -L/home/pierre/.pyenv/versions/3.8.2/lib -L/home/pierre/.pyenv/versions/3.8.2/lib build/temp.linux-x86_64-3.8/piconumpy/_piconumpy_cpython_capi.o -o build/lib.linux-x86_64-3.8/piconumpy/_piconumpy_cpython_capi.cpython-38-x86_64-linux-gnu.so
building 'piconumpy._piconumpy_hpy' extension
C compiler: gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC

creating build/temp.linux-x86_64-3.8/home
creating build/temp.linux-x86_64-3.8/home/pierre
creating build/temp.linux-x86_64-3.8/home/pierre/Dev
creating build/temp.linux-x86_64-3.8/home/pierre/Dev/hpy
creating build/temp.linux-x86_64-3.8/home/pierre/Dev/hpy/hpy
creating build/temp.linux-x86_64-3.8/home/pierre/Dev/hpy/hpy/devel
creating build/temp.linux-x86_64-3.8/home/pierre/Dev/hpy/hpy/devel/src
creating build/temp.linux-x86_64-3.8/home/pierre/Dev/hpy/hpy/devel/src/runtime
compile options: '-DHPY_UNIVERSAL_ABI -I/home/pierre/Dev/hpy/hpy/devel/include -I/home/pierre/.pyenv/versions/3.8.2/include/python3.8 -c'
extra options: '-Wfatal-errors -Werror'
gcc: piconumpy/_piconumpy_hpy.c
gcc: /home/pierre/Dev/hpy/hpy/devel/src/runtime/argparse.c
gcc -pthread -shared -L/home/pierre/.pyenv/versions/3.8.2/lib -L/home/pierre/.pyenv/versions/3.8.2/lib build/temp.linux-x86_64-3.8/piconumpy/_piconumpy_hpy.o build/temp.linux-x86_64-3.8/home/pierre/Dev/hpy/hpy/devel/src/runtime/argparse.o -o build/lib.linux-x86_64-3.8/piconumpy/_piconumpy_hpy.cpython-38-x86_64-linux-gnu.so
copying build/lib.linux-x86_64-3.8/piconumpy/_piconumpy_cpython_capi.cpython-38-x86_64-linux-gnu.so -> piconumpy
error: can't copy 'build/lib.linux-x86_64-3.8/piconumpy/_piconumpy_hpy.hpy.so': doesn't exist or not a regular file
```

Instead of creating build/lib.linux-x86_64-3.8/piconumpy/_piconumpy_hpy.hpy.so, it creates build/lib.linux-x86_64-3.8/piconumpy/_piconumpy_hpy.cpython-38-x86_64-linux-gnu.so.

Am I doing something wrong?

**Comment by antocuni** [comment]
> Am I doing something wrong?

You are doing the correct thing, the fact that it generates `_piconumpy_hpy.cpython-38-x86_64-linux-gnu.so` is a bug. You can work around the issue by manually renaming it do `.hpy.so` if you want to try.
Related issues: #85 and #86 

I will start a branch to fix it

**Comment by paugier** [comment]
Thank you for the explanation. 

I manage to import and bench the HPy implementation of piconumpy, with both ABI!
```
Julia                      :     1 * norm = 0.00204 s
Transonic-Pythran          : 0.576 * norm
Numpy                      :  34.9 * norm
PicoNumpy (purepy)         :  36.9 * norm
PicoNumpy (purepy_array)   :    35 * norm
PicoNumpy (Cython)         :  26.2 * norm
PicoNumpy (CPython C-API)  :  8.38 * norm
PicoNumpy (hpy)            :  8.63 * norm     (3% slower than PicoNumpy CPython C-API)
PicoNumpy (hpy universal)  :  11.5 * norm    (37% slower than PicoNumpy CPython C-API)
```

That's already great to be able to do that! Unfortunately, I can't import the universal piconumpy extension with pypy. I get

```
RPython traceback:
  File "pypy_interpreter.c", line 43326, in BuiltinCode1_fastcall_1
  File "pypy_module__hpy_universal.c", line 2940, in descr_load
  File "pypy_module__hpy_universal.c", line 10977, in HandleManager_close
  File "pypy_module__hpy_universal.c", line 10905, in HandleManager_close
Fatal RPython error: AssertionError
Aborted (core dumped)
```

I guess it is useless to open another issue here or on PyPy issue tracker.

**Comment by antocuni** [comment]
Fixed by #105 .

@paugier : I think that the latest version of hpy in the pypy py3k branch misses a couple of features which are neede to run piconumpy. I managed to run it correctly on pypy using revision 02f19eb67e30 (which is in the `hpy` branch).

In the [#hpy IRC logs](https://freenode.logbot.info/hpy/20201005#c5346388) you can find some of my comments about the results. Btw, you should join the channel to have more real time communications, if you want!
